docs/attendances_and_absences.md

```diff
@@ -0,0 +1,148 @@
+# Attendances & Absences
+
+## Definitions
+
+- absence  
+A day on which a child is scheduled to attend but does not - represented by a `ServiceDay` with an `absence_type`
+
+- attendance  
+A day on which a child attends, whether or not they are scheduled - represented by an `Attendance` with `check_in` and an optional `check_out`
+
+- `ServiceDay`  
+A record that `has_many` attendances (0 or more) and `belongs_to` a child and represents a single day in the child's timezone
+
+## Expected Behavior
+
+### Absences added during Onboarding 
+
+#### _Internal Onboarding CSV Import -  `Wonderschool::Necc::OnboardingCaseImporter`_
+
+- as a new approval is added, create a new absence for every child in that approval from the beginning of the approval to the end of the approval (or to the day of onboarding, whichever comes first)
+
+### Attendance Entry
+
+#### _Internal Pie CSV Import - `AttendanceCsvImporter`_
+
+- for each row:
+  - if the row contains an attendance
+    - if a `ServiceDay` exists for that day for that child
+      - add an `Attendance` w/ the `check_in`, `check_out` associated to that `ServiceDay`
+    - if a `ServiceDay` does not exist for that day for that child
+      - create a `ServiceDay` for that day for that child
+      - add an `Attendance` w/ the `check_in`, `check_out`, associated to the new `ServiceDay`
+  - if the row contains an absence
+    - if the child is scheduled for that day
+      - if a `ServiceDay` exists for that day for that child
+        - change the `ServiceDay` to an absence w/ the `absence_type`
+      - if a `ServiceDay` does not exist for that day for that child
+        - create a `ServiceDay` for that day for that child w/ the `absence_type`
+    - if the child is not scheduled for that day
+      - if a `ServiceDay` exists for that day for that child
+        - delete the `ServiceDay`
+      - if a `ServiceDay` does not exist for that day for that child
+        - do nothing
+
+    > Behavior currently not defined 
+    > - error handling
+
+#### _Daily Wonderschool CSV Import - `Wonderschool::Necc::AttendanceCsvImporter`_
+
+**Assumption: We do not get absences from Wonderschool**
+
+- for each row:
+  - if an attendance exists with the `wonderschool_id`
+    - update the attendance w/ the `check_in`, `check_out`
+  - if an attendance does not exist with the `wonderschool_id`
+    - if a `ServiceDay` exists for that day for that child
+      - add an `Attendance` w/ the `check_in`, `check_out`, `wonderschool_id` associated to that `ServiceDay`
+    - if a `ServiceDay` does not exist for that day for that child
+      - create a `ServiceDay` for that day for that child
+      - add an `Attendance` w/ the `check_in`, `check_out`, `wonderschool_id` associated to the new `ServiceDay`
+
+  > Behavior currently not defined 
+  > - if an attendance exists in the wonderschool import one day and does not exist the next day, we do not find missing attendances and remove them
```

Comment:
Decided with @katelovescode that we will not check for this behavior: "if an attendance exists in the wonderschool import one day and does not exist the next day"